cloud/aws/providerconfig/types.go

```diff
@@ -14,25 +14,140 @@
 package providerconfig
 
 import (
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// AWSMachineProviderConfig is the type that will be embedded in a Machine.Spec.ProviderConfig field
+// for an AWS instance. It is used by the AWS machine actuator to create a single machine instance,
+// using the RunInstances call (https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_RunInstances.html)
+// Required parameters such as region that are not specified by this configuration, will be defaulted
+// by the actuator.
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 type AWSMachineProviderConfig struct {
-	metav1.TypeMeta `json:",inline"`
+	metav1.TypeMeta
+
+	// AMI is the reference to the AMI from which to create the machine instance.
+	AMI AWSResourceReference
+
+	// InstanceType is the type of instance to create. Example: m4.xlarge
+	InstanceType string
+
+	// AdditionalTags is the set of tags to add to an instance, in addition to the ones
+	// added by default by the actuator. These tags are additive. The actuator will ensure
+	// these tags are present, but will not remove any other tags that may exist on the
+	// instance.
+	// +optional
+	AdditionalTags map[string]string
+
+	// IAMInstanceProfile is a reference to an IAM role to assign to the instance
+	// +optional
+	IAMInstanceProfile *AWSResourceReference
+
+	// PublicIP specifies whether the instance should get a public IP.
+	// Precedence for this setting is as follows:
+	// 1. This field if set
+	// 2. Cluster/flavor setting
+	// 3. Subnet default
+	// +optional
+	PublicIP *bool
+
+	// SecurityGroups is an array of references to security groups that should be applied to the
+	// instance. These security groups would be set in addition to any security groups defined
+	// at the cluster level or in the actuator.
+	// +optional
+	SecurityGroups []AWSResourceReference
```

Comment:
Can this be renamed to AdditionalSecurityGroups to align with the updated comments?